### DISCUSSION TITLE
Window Throw: Different settings for portrait vs landscape monitors

### DISCUSSION THREAD

**Discussion by tmchow** [rxhanson/RectanglePro-Community#734] [General]
Would be useful to have option to have different window throw behavior for portrait vs landscape monitors.

![image](https://user-images.githubusercontent.com/517103/217379778-6e3e9614-9ccf-4df4-83e2-911fd951ca1a.png)

For example, on portrait monitors, 1/3 vs 2/3 options (vertically) are more common to use than in landscape monitors for obvious reasons. Similar to how horizontal 1/3 options are less common if ever used on portrait monitors by vast majority of people.

**Comment by rxhanson** [comment]
I have a couple of backlog items for handling something like this:

1. Window throw profiles, with display as a differentiator.
2. Custom shortcuts that have both landscape and portrait size/position fields, where the proper size/position is applied depending on the display. 

Both of these have pros and cons, and right now option 2 is higher in priority because it gives a lot more flexibility (at the cost of more involved configuration, unfortunately), and it will take less time to implement.

**Comment by tmchow** [comment]
I think both options will work for what I'm asking for

**Comment by GlacierWalrus** [comment]
> I have a couple of backlog items for handling something like this:
> 
> 1. Window throw profiles, with display as a differentiator.
> 2. Custom shortcuts that have both landscape and portrait size/position fields, where the proper size/position is applied depending on the display.
> 
> Both of these have pros and cons, and right now option 2 is higher in priority because it gives a lot more flexibility (at the cost of more involved configuration, unfortunately), and it will take less time to implement.


How are these coming along?

I'm trying out rectangle pro and I _love_ it so far, but my setup is one horizontal centre display and one vertical on either side, and throwing not being orientation aware means it doesn't work well for me.

Are there any workarounds, or do I have to sacrifice my layout preference for either horizontal or vertical to use throws?

**Reply by rxhanson** [reply]
Unfortunately, as you're aware, these haven't made it out the door. Still commonly asked for, and still in the queue. 

At the moment the only workaround is to make use of more shortcuts - long throw and quick throw. Not ideal and hopefully I can get to this soon.

**Reply by bpkroth** [reply]
Is there a community source code version of this that I could use to submit a PR for this? In the age of agents, I am happy to donate a few my own token credits at this to see it fixed :)

**Reply by rxhanson** [reply]
I've just outlined the current way to do this on this thread (https://github.com/rxhanson/RectanglePro-Community/discussions/734#discussioncomment-18005313). 

Thanks for offering! This part is not open source, and even so I have some specific ways that I'd like this implemented to facilitate more window throw configuration, like having additional ones with different modifiers, etc. On top of that I am pretty particular about the code that goes into Rectangle Pro :)

I've recently maneuvered myself into a position where I have significantly more time to work on all of my apps, so we'll see a lot more of these backlog items getting attention going forward.

**Comment by rxhanson** [comment]
In v3.82, I added the ability to specify the Origin Display on Custom Size & Position entries, and this can be used to build out different settings for portrait vs landscape.

I realize it can be a little tedious configuring it this way, but I chose to implement this way first because it provides a lot of flexibility that's already built into the app. I'll add this to the guide soon with more details, but here's a quick example:

First create a custom size and position, and either select from a preset or set your own, then add a per-display size/position:

<img width="956" height="718" alt="Screenshot 2026-08-13 at 1 30 35 PM" src="https://github.com/user-attachments/assets/d8854c14-b752-4e6c-90f6-75d889ad8604" />

<img width="874" height="718" alt="custom" src="https://github.com/user-attachments/assets/b578e001-43be-4670-a32c-110f7f7a9a51" />

Next go back to the window throw and select it:

<img width="917" height="1032" alt="window throw" src="https://github.com/user-attachments/assets/8d3ca505-0c18-454c-b4b6-1de8f786054a" />

I still have plans to create a more streamlined and straightforward way to do this, but at least it's possible in its current form.
